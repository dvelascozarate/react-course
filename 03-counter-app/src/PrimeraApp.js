import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({ nombre, subtitulo }) => {

    const saludo = `Hola ${nombre}`;

    return <><h1>{saludo}</h1><p>{subtitulo}</p></>;
};

PrimeraApp.propTypes = {
    nombre: PropTypes.string.isRequired,
    subtitulo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    nombre: 'Mundo',
    subtitulo: 'Subtitulo'
}

export default PrimeraApp;