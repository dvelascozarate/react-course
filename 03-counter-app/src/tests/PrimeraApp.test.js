import React from 'react';
import { render } from "@testing-library/react"
import "@testing-library/jest-dom"
import PrimeraApp from '../PrimeraApp'
import { shallow } from 'enzyme';

describe('Pruebas en el componente PrimeraApp', () => {
    // Prueba utilizando los mÃ©todos de prueba de react    
    test('Debe renderizar correctament el mensaje', () => {

        const nombre = 'Daniel'
        const { getByText } = render(<PrimeraApp nombre={nombre} />);

        expect(getByText(`Hola ${nombre}`)).toBeInTheDocument();
    })

    // Prueba utilizando Enzyme

    test('Debe de mostrar el componente correctamente', () => {

        const nombre = 'Daniel';
        const wrapper = shallow(<PrimeraApp nombre={nombre} />);

        expect(wrapper).toMatchSnapshot();

    })

    test('Debe mostrar correctamente el subtitulo', () => {
        const nombre = 'Daniel';
        const subtitulo = 'Soy un subtitulo';
        const wrapper = shallow(<PrimeraApp nombre={nombre} subtitulo={subtitulo} />);

        const texto = wrapper.find('p').text();

        expect(texto).toBe(subtitulo);

    })



})
