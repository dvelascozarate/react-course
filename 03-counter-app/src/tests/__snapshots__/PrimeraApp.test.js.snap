// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente PrimeraApp Debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h1>
    Hola Daniel
  </h1>
  <p>
    Subtitulo
  </p>
</Fragment>
`;
